### PR TITLE
Enabled Mellanox CI for release branches (changes for v4.0.x branch).

### PR DESCRIPTION
**Pull request by artemry-mlnx**
Signed-off-by: Artem Ryabov <artemry@mellanox.com>

CC @amaslenn

### CODE REVIEW

**Comment by jsquyres** [comment]
ok to test

**Comment by artemry-mlnx** [comment]
@jsquyres 
Could you please merge this if there aren't any additional remarks?

File: .ci/mellanox/azure-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+trigger:
```
Comment (mike-dubman) [inline .ci/mellanox/azure-pipelines.yml]: @artemry-mlnx - no need trigger on "merge" event into branches. only on PR.

**Comment by hppritcha** [comment]
@rhc54 we wanted to wait on this till after the 4.0.3 release.

**Comment by rhc54** [comment]
My apologies - I didn't realize that. Please feel free to revert.

**Comment by rhc54** [comment]
Clarifying: I didn't realize it was a release branch or I wouldn't have committed it. I thought since they were asking @jsquyres to commit it, we were talking about master.

File: .ci/mellanox/azure-pipelines.yml (hunk 1 above)
Comment (artemry-mlnx) [reply .ci/mellanox/azure-pipelines.yml]: Fixed in https://github.com/open-mpi/ompi/pull/7465

**Pull request merged**